core/src/main/java/com/linecorp/armeria/client/AbstractHttpResponseDecoder.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import java.util.Iterator;
+
+import com.linecorp.armeria.common.ContentTooLargeException;
+import com.linecorp.armeria.common.ContentTooLargeExceptionBuilder;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.internal.client.DecodedHttpResponse;
+import com.linecorp.armeria.internal.client.HttpSession;
+import com.linecorp.armeria.internal.common.InboundTrafficController;
+import com.linecorp.armeria.internal.common.KeepAliveHandler;
+
+import io.netty.channel.Channel;
+import io.netty.channel.EventLoop;
+import io.netty.util.collection.IntObjectHashMap;
+import io.netty.util.collection.IntObjectMap;
+
+abstract class AbstractHttpResponseDecoder implements HttpResponseDecoder {
```

Comment:
`HttpResponseDecoder` now becomes `AbstractHttpResponseDecoder` and the new interface `HttpResponseDecoder` is added for `WebSocketHttp1ClientChannelHandler`.